Lesson22/nurbasak/script.js

```diff
@@ -0,0 +1,188 @@
+const emailPattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+const phonePattern = /^\+?\d(?:\s?\d){9,19}$/;
+const onlyLetters = /^[a-zA-Z\s-]+$/;
+const cardPattern = /^(\d{4}\s){3}\d{4}$/;
+const expDatePattern = /^(0[1-9]|1[0-2])\/(2[5-9]|[3-9]\d)$/; // Revised this pattern after a little research to 2025-2099.
+const cvvPattern = /^\d{3}$/; // Revised this pattern after a little research to 3 or 4 digits.
+
+const emailInput = document.getElementById("email");
+const emailError = document.getElementById("emailError");
+const phoneInput = document.getElementById("phone");
+const phoneError = document.getElementById("phoneError");
+
+const firstNameInput = document.getElementById("firstname");
+const firstNameError = document.getElementById("firstNameError");
+const lastNameInput = document.getElementById("lastname");
+const lastNameError = document.getElementById("lastNameError");
+const cardInput = document.getElementById("cardNumber");
+const cardNumberError = document.getElementById("cardNumberError");
+
+const expDateInput = document.getElementById("expDate");
+const expDateError = document.getElementById("expDateError");
+const cvvInput = document.getElementById("cvv");
+const cvvError = document.getElementById("cvvError");
+const inputGroups = document.querySelectorAll(".input-group");
+const form = document.getElementById("checkoutForm");
+const submitButton = document.getElementById("submitBtn");
+
+const successMessage = document.getElementById("success");
+
+let formValid = true;
+
+function validateEmail(email) {
+  if (!emailPattern.test(email)) {
+    emailError.innerText =
+      "Please enter email address in the correct format (e.g. name@gmail.com)";
+    emailError.classList.remove("hidden");
+    formValid = false;
+  } else {
+    emailError.innerText = "";
+    emailError.classList.add("hidden");
+  }
+}
+
+function validatePhone(phone) {
+  if (!phonePattern.test(phone)) {
+    phoneError.innerText =
+      "Please enter phone in the following format: +90 123 456 7890";
```

Comment:
Your regex pattern is flexible and correctly supports both local and international formats. However, the current error message suggests that only the international format (+90) is valid.
To avoid confusion and match the flexibility of your code, it is better to include both examples in the error message.

```suggestion
      "Please enter a valid phone number (e.g. 555 123 4567 or +90 555 123 4567)";
```